app/src/main/java/com/github/orkest/domain/FireStoreDatabaseAPI.kt

```diff
@@ -165,17 +163,31 @@ class FireStoreDatabaseAPI {
 
     //===========================SONG POST OPERATIONS======================
 
-    fun storeTokenInDatabase(username:String,token:String?): CompletableFuture<Boolean>{
+    fun storeDeezerInformationsInDatabase(username:String,token:String?,userId:String?="",playlistId:String=""): CompletableFuture<Boolean>{
         val completableFuture = CompletableFuture<Boolean>()
         val path = "deezerToken"
-        db.collection(path).document(username).set(hashMapOf ("token" to token,"id" to "","Playlist_Id" to "")).addOnSuccessListener {
+        db.collection(path).document(username).set(DeezerInformations(token,userId,playlistId)).addOnSuccessListener {
             completableFuture.complete(true)
         }.addOnFailureListener{
             completableFuture.complete(false)
         }
         return completableFuture
     }
 
+    fun getUserDeezerInformations(username: String):CompletableFuture<DeezerInformations>{
+        val completableFuture = CompletableFuture<DeezerInformations>()
+        val path = "deezerToken"
+        db.collection(path).document(username).get().addOnSuccessListener {
+            completableFuture.complete(it.toObject(DeezerInformations::class.java))
+        }.addOnFailureListener{
+            completableFuture.complete(DeezerInformations("","",""))
+        }
+        return completableFuture
+    }
```

Comment:
Maybe use a constant for "deezerToken" instead of hardcoding it.

This remark depends on how the rest was coded but I think it would better if instead of an empty DeezerInformations if there's an error, we would send the completable future with the exception, like :

.addOnFailureListener { e ->
            completableFuture.completeExceptionally(e)
        }
       
This could be a better way of handling the exception.

Reply:
True will try to fix in a following commit,